felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -505,6 +506,22 @@ func newBPFEndpointManager(
 		)
 		m.services = make(map[serviceKey][]ip.CIDR)
 		m.dirtyServices = set.New[serviceKey]()
+		m.natExcludedIPs = ip.NewCIDRTrie()
+
+		var excludeIPsMatch = 1
+
+		for _, c := range config.BPFExcludeIPsFromNAT {
+			cidr, err := ip.CIDRFromString(c)
+			if err != nil {
+				log.WithError(err).Warnf("Bad %s CIDR to exclude from NAT", c)
+			}
+
+			if int(cidr.Version()) == 6 && !m.ipv6Enabled {
```

Comment:
What about the reverse, IPv4 CIDR and IPv6 support enabled?  
```suggestion
			if (cidr.Version() == 6) != m.ipv6Enabled {
```
do you need the cast?

Reply:
Atm if `m.ipv6Enabled` then ipv4 should not be considered. As part of the dual stack effort, this will need to be addressed, but idk how that is going to pan out yet.